cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/Labels.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * An immutable set of labels
+ */
+public class Labels {
+
+    public static final String STRIMZI_DOMAIN = "strimzi.io/";
+
+    /**
+     * The kind of a ConfigMap:
+     * <ul>
+     *     <li>{@code strimzi.io/kind=cluster}
```

Comment:
Technically speaking this is not true, or? This is actually only the value used in our example files. But it can be easily changed through the environment variable. So I'm not sure whether this should be in the Javadoc like this.

(TBH looking back I'm also not sure whether we should give so much freedom, because it creates complications for documentation etc. But that goes beyond this PR.)

Reply:
There are problems with that (as you know) when it comes to documentation etc. Specifically for `TopicController.defaultTopicConfigMapLabels()` we have to formulate the labels which the TC will watch for, currently we do:

```java
String.format("%s=%s,%s=%s",
                Labels.STRIMZI_CLUSTER_LABEL, cluster,
                Labels.STRIMZI_KIND_LABEL, TopicController.TOPIC_CM_KIND);
```

If `STRIMZI_KIND_LABEL` is not a constant we will have to pass whatever the label key is into that method. But if that label key is something that the user "owns" and they configure `foo=KLUSTER`, who are we to go forming `foo=topic` here and saying that should mean something? Surely they own the `foo` space!

Since it's not clear that configurability of the kind label is actually needed by users, I think it would make life much simpler for us to take ownership of the label used for the watches and say it's always `strimzi.io/kind` and it's `=cluster` for CMs consumed by the CC and `=topic` for CMs consumed by the TC.

Reply:
Yeah, indeed. I do not think there is any value in the user changing our `strimzi.io/*` labels. 

There is a value for the user to attach additional labels (for example for cost tracking, security purposes etc.). These should not be needed for us to search for the resources. We just need to copy them to everything we create. But this doesn't affect the `strimzi.io/*` labels.

So IMHO the only question is whether we change it now or later :-).

Reply:
For what it's worth ... I agree with you both guys ;)